lib/Controller/PageController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+// SPDX-FileCopyrightText: Hamza Mahjoubi <hamzamahjoubi22@proton.me>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+namespace OCA\Files_EmailViewer\Controller;
+
+use OCA\Files_EmailViewer\AppInfo\Application;
+use OCA\Files_EmailViewer\Exception\ConversionException;
+use OCA\Files_EmailViewer\Service\ConversionService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\ContentSecurityPolicy;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Http\StreamResponse;
+use OCP\Files\IRootFolder;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class PageController extends Controller {
+	public function __construct(
+		IRequest                  $request,
+		private IRootFolder          $root,
+		private ?string           $userId,
+		private ConversionService $conversionService,
+		private LoggerInterface $logger
+	) {
+		parent::__construct(Application::APP_ID, $request);
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 * @return StreamResponse|DataResponse
+	 */
+	public function convert(string $fileId) {
```

Comment:
minor limitation: we can't use this approach in Mail because attachments are not on the Nextcloud file system

but viewing in Files is fine for now :+1: